src/ui/public/sort/sort_properties.js

```diff
@@ -0,0 +1,86 @@
+import _ from 'lodash';
+
+/**
+ * @typedef {Object} SortProperty
+ * @property {string} sortProperty.name - Name of the property.
+ * @property {function} sortProperty.getValue - A function that takes in an object and returns a value to sort
+ * by.
+ * @property {boolean} sortProperty.isAscending - The direction of the last sort by this property. Used to preserve
+ * past sort orders.
+ */
+
+/**
+ * Stores sort information for a set of SortProperties, including which property is currently being sorted on, as
+ * well as the last sort order for each property.
+ */
+export class SortProperties {
```

Comment:
Whoa I could have sworn I added a comment that said I was going to add tests as soon as you approved of the general direction.  :)  Just wanted to make sure we were on the same page first, in case there was conflict and the pr got scrapped.